module/documents/actor/actor.mjs

```diff
@@ -2870,7 +2870,17 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
       if ( settings.keep.has("languages") ) d.system.traits.languages = o.system.traits.languages;
 
       // Keep specific items from the original data
-      d.items = d.items.concat(o.items.filter(i => {
+      const profDiff = source.system.attributes.prof - this.system.attributes.prof;
+      const numberFormat = new Intl.NumberFormat(game.i18n.lang, { signDisplay: "always" });
+      d.items = d.items.map(i => {
+        if ( settings.keep.has("class") && ((i.type === "feat") || (i.type === "weapon")) ) {
+          // Items gained from the source should use the source's proficiency bonus.
+          Object.values(i.system.activities).forEach(activity => {
+            if ( activity.type === "attack" ) activity.attack.bonus += numberFormat.format(profDiff);
```

Comment:
Seems like this could result in some numbers we cannot parse depending on the language the user has selected:

```javascript
(new Intl.NumberFormat("ne", { signDisplay: "always" })).format(5) // Result: +५
// (for all our Nepali users)
```